numpy/f2py/tests/test_crackfortran.py

```diff
@@ -88,6 +88,29 @@ def test_defaultPublic(self, tmp_path):
         assert 'private' not in mod['vars']['seta']['attrspec']
         assert 'public' in mod['vars']['seta']['attrspec']
 
+    def test_inlineDefinition(self, tmp_path):
+        f_path = tmp_path / "mod.f90"
+        with f_path.open('w') as ff:
+            ff.write(textwrap.dedent("""\
+            module foo
+              public
+              type, private, bind(c) :: a
+                integer :: i
+              end type a
+              type, bind(c) :: b
+                integer :: j
+              end type b
+              public :: b
+            end module foo
+            """))
+        mod = crackfortran.crackfortran([str(f_path)])
+        assert len(mod) == 1
+        tt = mod[0]['vars']
+        assert 'private' in tt['a']['attrspec']
+        assert 'public' not in tt['a']['attrspec']
+        assert 'public' in tt['b']['attrspec']
+        assert 'private' not in tt['b']['attrspec']
```

Comment:
Replace with
```python
assert set(tt['a']['attrspec']) == {'private', 'bind(c)'}
assert set(tt['b']['attrspec']) == {'public', 'bind(c)'}
```

Reply:
Much more compact way of doing more !

I've added a other case for:
```
type :: c
  integer :: k
end type c
```
where there is no attributes.